packages/3dom/src/utilities.ts

```diff
@@ -22,3 +23,176 @@ export const getLocallyUniqueId = (() => {
   let id = 0;
   return () => id++;
 })();
+
+/**
+ * A VisitorCallback is used to access an element of a GLTF scene graph. The
+ * callback receives a reference to the element, the element's index relative
+ * to other elements of its type (as it would appear in the top-level element
+ * array in the corresponding GLTF) and a hierarchy array that is the list of
+ * ancestor elements (inclusive of the current element) that preceded the
+ * currently accessed element in the GLTF scene graph hierarchy.
+ */
+export type VisitorCallback<T extends GLTFElement> =
+    (element: T, index: number, hierarchy: GLTFElement[]) => void;
+
+/**
+ * There is a corresponding VisitorCallback type for every type of GLTFElement.
+ */
+export type VisitorCallbacks = {
+  [T in keyof GLTFElementMap]?: VisitorCallback<GLTFElementMap[T]>;
+}
+
+/**
+ * Supported options for configuring a GLTFTreeVisitor include:
+ *  - allScenes: if true, all scenes (not just the active one) will be visited
```

Comment:
In this case, I am usually the word active casually. The glTF 2.0 specification currently defines a single, top-level `scene` field that we always assume is the "active" scene in `<model-viewer>`. We currently do not have a mechanism for activating other scenes, or any semantics for what that means, so that stuff is all TBD.

Reply:
gotcha; a note to that effect will probably help me someday :)